packages/headless/src/controllers/commerce/search/facets/headless-search-facet-options.ts

```diff
@@ -0,0 +1,35 @@
+import {isFacetResponse} from '../../../../features/commerce/facets/facet-set/facet-set-selector';
+import {executeSearch} from '../../../../features/commerce/search/search-actions';
+import {
+  CommerceFacetSetSection,
+  CommerceSearchSection,
+} from '../../../../state/state-sections';
+import {CoreCommerceFacetOptions} from '../../facets/core/headless-core-commerce-facet';
+
+const facetResponseSelector = (
+  state: CommerceSearchSection & CommerceFacetSetSection,
+  facetId: string
+) => {
+  const response = state.commerceSearch.facets.find(
+    (response) => response.facetId === facetId
+  );
+  if (isFacetResponse(state, response)) {
+    return response;
+  }
+
+  return undefined;
+};
+
+const isFacetLoadingResponseSelector = (state: CommerceSearchSection) =>
+  state.commerceSearch.isLoading;
+
+export const commonOptions: Pick<
+  CoreCommerceFacetOptions,
+  | 'fetchResultsActionCreator'
+  | 'facetResponseSelector'
+  | 'isFacetLoadingResponseSelector'
+> = {
+  fetchResultsActionCreator: executeSearch,
+  facetResponseSelector,
+  isFacetLoadingResponseSelector,
+};
```

Comment:
This logic is very similar to the one for `ehadless-product-listing-facet-options.ts`. Maybe in the future, instead of having two selectors, we could have a single `getResponseState` selector which results `{ isLoading: boolean, facets: AnyFacetResponse[] }`